### ISSUE TITLE
Limited food seems to prefer long teeth

### ISSUE THREAD

**Issue by amanda-phet** [phetsims/natural-selection#161]
Related to #158 maybe. Just an extension of the limited food issue. 

I am experimenting with Limited Food and different mutations. When I add the long teeth mutation, the log will frequently read out `No bunnies with long teeth were starved because the population is <= 4.` but teeth should not matter for limited food. This happens whether long teeth are dominant or recessive. 

You can see in the graph during generation 2 here, none of the bunnies with long teeth died, and the log confirmed it.

<img width="460" alt="image" src="https://user-images.githubusercontent.com/7852916/89447804-d9ffec80-d713-11ea-8035-b295691daca7.png">

Also here in generations 4, 5, and 8. 
<img width="458" alt="image" src="https://user-images.githubusercontent.com/7852916/89448015-33681b80-d714-11ea-90be-bbfd365de8ac.png">

**Comment by amanda-phet** [comment]
For https://github.com/phetsims/QA/issues/522

**Comment by amanda-phet** [comment]
Using Mac, Chrome, and rc.1

**Comment by pixelzoom** [comment]
It would be helpful to have the log and any query parameters that you used.

**Comment by pixelzoom** [comment]
Since @amanda-phet noted "For phetsims/QA#522" I'm going to assume that this is blocking for the 1.0 prototype. Labeling as "blocks-sim-publication".  If that's incorrect, let me know.

**Comment by amanda-phet** [comment]
No need to add a mutation for teeth. Here is the log after adding a mate, turning on limited food at the start of generation 3: 

====== Generation 1 ======
natural-selection_all_phet.html?log:81 0 bunnies died of old age
natural-selection_all_phet.html?log:81 4 bunnies were born
natural-selection_all_phet.html?log:81 attempting to starve bunnies
natural-selection_all_phet.html?log:81 total live bunnies = 6
natural-selection_all_phet.html?log:81 total dead bunnies = 0
natural-selection_all_phet.html?log:81 ====== Generation 2 ======
natural-selection_all_phet.html?log:81 0 bunnies died of old age
natural-selection_all_phet.html?log:81 using bunnyRestRange=[2,7] for population=10
natural-selection_all_phet.html?log:81 12 bunnies were born
natural-selection_all_phet.html?log:81 attempting to starve bunnies
natural-selection_all_phet.html?log:81 total live bunnies = 18
natural-selection_all_phet.html?log:81 total dead bunnies = 0
natural-selection_all_phet.html?log:81 ====== Generation 3 ======
natural-selection_all_phet.html?log:81 0 bunnies died of old age
natural-selection_all_phet.html?log:81 36 bunnies were born
natural-selection_all_phet.html?log:81 attempting to starve bunnies
**natural-selection_all_phet.html?log:81 No bunnies with long teeth were starved because the population is <= 4.**
natural-selection_all_phet.html?log:81 36 bunnies with short teeth died of starvation

The examples above show the concern, which is that someone might think limited food selects for a certain phenotype when it should not.

**Comment by pixelzoom** [comment]
Yes, limited food will prefer long teeth in specific circumstances, and that's what you requested.

In https://github.com/phetsims/natural-selection/issues/153, you asked to ignore limited food if the total population (not the population of a specific phenotype) is < 7.  It's not in this case, so limited food will be applied.

In https://github.com/phetsims/natural-selection/issues/98#issuecomment-646275437, you asked for:
>  Have the selection factor ignore preferred phenotype if fewer than N exist

This constraint is specific to the "long teeth" phenotype.  In your example, the number of bunnies with long teeth is <= 4, so they are not affected by limited food, and are therefore "preferred"

Does that make sense? Or did I misunderstand the requirements?

**Comment by amanda-phet** [comment]
>This constraint is specific to the "long teeth" phenotype, and we settled on N <= 4. In your example, the number of bunnies with long teeth is <= 4, so they are not affected by limited food, and are therefore "preferred"

The constraint of N <= 4 was for long teeth (when tough food is checked) or fur color (when wolves are checked). Since limited food doesn't have a preferred phenotype (and the model clearly reflects this when the population is large, regardless of mutations), it should not prefer long teeth at any time.

**Comment by amanda-phet** [comment]
I said 
>The constraint of N <= 4 was for long teeth (when tough food is checked) or fur color (when wolves are checked).

And it looks like for fur color the constraint is <= 5. I am starting to get confused about this, because I thought we picked one value for this constraint. I'll need to review #98 for this. 

However, we still need to do something about limited food, because this constraint should not apply.

**Comment by amanda-phet** [comment]
I reviewed the mutation model and found https://github.com/phetsims/natural-selection/issues/98#issuecomment-656438400, which is helpful to confirm that I indeed picked different values for the constraint. That's fine that the values are different, but I should have reviewed the model with `?log` turned on way back when I approved it, because I didn't notice this issue (that limited food prefers long teeth when N≤4) until just now when watching the log more closely. Limited food works fine with larger populations, and doesn't care about teeth or any other traits, so we just need to fix the constraint.
